optaplanner-benchmark/src/main/java/org/optaplanner/benchmark/config/PlannerBenchmarkConfig.java

```diff
@@ -279,32 +279,9 @@ protected int resolveParallelBenchmarkCount() {
         if (parallelBenchmarkCount == null) {
             resolvedParallelBenchmarkCount = 1;
         } else if (parallelBenchmarkCount.equals(PARALLEL_BENCHMARK_COUNT_AUTO)) {
-            // TODO Tweak it based on experience
-            if (availableProcessorCount <= 2) {
-                resolvedParallelBenchmarkCount = 1;
-            } else if (availableProcessorCount <= 4) {
-                resolvedParallelBenchmarkCount = 2;
-            } else {
-                resolvedParallelBenchmarkCount = (availableProcessorCount / 2) + 1;
-            }
+            resolvedParallelBenchmarkCount = resolveParallelBenchmarkCountAutomatically(availableProcessorCount);
         } else {
-            String scriptLanguage = "JavaScript";
-            ScriptEngine scriptEngine = new ScriptEngineManager().getEngineByName(scriptLanguage);
-            scriptEngine.put(AVAILABLE_PROCESSOR_COUNT, availableProcessorCount);
-            Object scriptResult;
-            try {
-                scriptResult = scriptEngine.eval(parallelBenchmarkCount);
-            } catch (ScriptException e) {
-                throw new IllegalArgumentException("The parallelBenchmarkCount (" + parallelBenchmarkCount
-                        + ") is not " + PARALLEL_BENCHMARK_COUNT_AUTO + " and cannot be parsed in " + scriptLanguage
-                        + " with the variables ([" + AVAILABLE_PROCESSOR_COUNT + "]).", e);
-            }
-            if (!(scriptResult instanceof Number)) {
-                throw new IllegalArgumentException("The parallelBenchmarkCount (" + parallelBenchmarkCount
-                        + ") is resolved to scriptResult (" + scriptResult + ") in " + scriptLanguage
-                        + " and is not a Number.");
-            }
-            resolvedParallelBenchmarkCount = ((Number) scriptResult).intValue();
+            resolvedParallelBenchmarkCount = resolveParallelBenchmarkCountFromFormula(availableProcessorCount);
```

Comment:
I've been changing this code on my Partionted Search fork in the past few weeks, so I need to do figure this one out. Maybe I can try to get teh Partitioned Search fork in a masked-out appraoch so I can merge it in master and handle this PR, as I presume otherwise it will take too long for this PR to get merged.